src/Compilers/CSharp/Portable/Utilities/TypeSymbolExtensions.cs

```diff
@@ -96,14 +107,31 @@ public static bool HasCustomModifiers(this TypeSymbol type, bool flagNonDefaultA
                     {
                         var array = (ArrayTypeSymbol)type;
                         TypeWithAnnotations elementType = array.ElementTypeWithAnnotations;
-                        return elementType.CustomModifiers.Any() || elementType.Type.HasCustomModifiers(flagNonDefaultArraySizesOrLowerBounds) ||
-                               (flagNonDefaultArraySizesOrLowerBounds && !array.HasDefaultSizesAndLowerBounds);
+                        return checkTypeWithAnnotations(elementType) || (flagNonDefaultArraySizesOrLowerBounds && !array.HasDefaultSizesAndLowerBounds);
                     }
                 case SymbolKind.PointerType:
                     {
                         var pointer = (PointerTypeSymbol)type;
                         TypeWithAnnotations pointedAtType = pointer.PointedAtTypeWithAnnotations;
-                        return pointedAtType.CustomModifiers.Any() || pointedAtType.Type.HasCustomModifiers(flagNonDefaultArraySizesOrLowerBounds);
+                        return checkTypeWithAnnotations(pointedAtType);
+                    }
+                case SymbolKind.FunctionPointer:
+                    {
+                        var funcPtr = (FunctionPointerTypeSymbol)type;
+                        if (checkTypeWithAnnotations(funcPtr.Signature.ReturnTypeWithAnnotations))
+                        {
+                            return true;
+                        }
+
+                        foreach (var param in funcPtr.Signature.Parameters)
+                        {
+                            if (checkTypeWithAnnotations(param.TypeWithAnnotations))
+                            {
+                                return true;
+                            }
+                        }
+
+                        return false;
```

Comment:
Consider using:
```
return funcPtr.Signature.Parameters.Any(
    (p, f) => checkTypeWithAnnotations(p, f),
    flagNonDefaultArraySizesOrLowerBounds);
```

Reply:
These methods can be called on very deeply nested generics. I'd prefer to avoid any kind of capturing.

---
In reply to: [422382953](https://github.com/dotnet/roslyn/pull/44078#discussion_r422382953) [](ancestors = 422382953)

Reply:
I don't think we should capture any state either. `checkTypeWithAnnotations` should be `static`.

---
In reply to: [423324444](https://github.com/dotnet/roslyn/pull/44078#discussion_r423324444) [](ancestors = 423324444,422382953)

Reply:
This needs to check both the TypeWithAnnotations and the ref custom modifiers, and I'm not planning on refactoring existing code beyond what I have to.

---
In reply to: [423326618](https://github.com/dotnet/roslyn/pull/44078#discussion_r423326618) [](ancestors = 423326618,423324444,422382953)